dcore/configuration.py

```diff
@@ -0,0 +1,81 @@
+"""
+Some simple tools for making model configuration nicer.
+"""
+
+
+class Configuration(object):
+    def __init__(self, **kwargs):
+
+        self.__setattr__ = self._proto__setattr__
+        for name, value in kwargs.iteritems():
+            print name, value
```

Comment:
So by "the code", I mean the file that you actually ran to produce the results.  I agree that the filename is the wrong place to put this.  The `State` object knows what all the user-specifiable configuration slots are.  So I think the correct thing to do is for `State` to gain a `dump` / `load` pair that you can either use to save the configuration used for a particular simulation, or else to load a configuration from a given save file.  I would be tempted to use JSON, rather than pickling, because then at least the dumped configuration is human readable.